### PR TITLE
Add shared product cloud docs

### PR DESCRIPTION
**Pull request by matkuliak**
## Checklist

 - [ ] Link to issue this PR refers to (if applicable): 
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by proddata** [comment]
Hi @matkuliak 

Wasn't aware, that you are working on this. With the restructuring of the docs I think it is worth to have a top-level service offerings page, which I started here: https://github.com/crate/cloud-docs/pull/35
Let's bring this two pages together.

File: docs/reference/subscription-plans.rst [hunk 1]
```diff
@@ -4,15 +4,14 @@
 Subscription Plans
 ==================
 
-When signing up for the CrateDB Cloud service, you have a choice of six
-different subscription plans. Each plan is preconfigured for different use
-cases, depending on what your storage, and computation needs are. At the same
+When signing up for the CrateDB Cloud service, you have a choice of
+different subscription plans. Each plan is preconfigured for different use cases, depending on your storage and computation needs. At the same
 time, the plans also offer elasticity, to accompany use case changes.
```
Comment (SStorm) [inline docs/reference/subscription-plans.rst:9]:
```suggestion
time, the plans also offer elasticity to accompany use case changes.
```

File: docs/reference/subscription-plans.rst [hunk 2]
```diff
@@ -4,15 +4,14 @@
...
 
-Within a given plan, you can horizontally scale the capacity of your
-cluster up or down by adding or subtracting nodes. Limited vertical 
-scaling is also possible, with storage scaling (up only). For more
-information on how to do this, please refer to the `scaling guide`_.
+Within certain plans, you can horizontally scale the capacity of your
+cluster up or down by adding or subtracting nodes. Limited vertical scaling is
+also possible with storage scaling (up only). For more information on how to
+do this, please refer to the `scaling guide`_.
```
Comment (SStorm) [inline docs/reference/subscription-plans.rst:14]:
```suggestion
do this, please refer to the `scaling guide`_. You can change plans at any time.
```

File: docs/reference/subscription-plans.rst [hunk 3]
```diff
@@ -71,46 +68,68 @@ Each plan has two dimensions: compute and storage.
...
+Shared
+------
 
-These are the base plans that offer the best performance, and the biggest
-flexibility. Their specs start from 1.75 CPU, 7 GiB RAM, and 32GiB storage, all
-the way to 14 CPU, 55 GiB RAM, and 512 GiB storage per-node. These plans can 
-be scaled to up to 9 nodes. If you need more nodes than this, feel free to
-`contact us`_ at any time.
+Shared-resource plans are used to deploy clusters that allow for better 
+resource utilization. These clusters share compute and storage resources
```
Comment (SStorm) [inline docs/reference/subscription-plans.rst:94]:
```suggestion
Shared-resource plans are used to deploy clusters that share compute and storage resources
```

I don't think "better resource utilization" is really a thing (except for us!)

File: docs/reference/subscription-plans.rst [hunk 4]
```diff
@@ -71,46 +68,68 @@ Each plan has two dimensions: compute and storage.
...
+with other clusters in this category. Because of this, they offer a more
+cost-effective solution for smaller teams and experimental deployments of
+low-traffic applications. They are limited to one node with up to 8 CPUs, 12 
+GiB of memory, and 1 TiB of storage. 
+
+The vCPU displayed for each tier is an 'up to' value and is not guaranteed.
+Memory and storage are guaranteed. Scaling within the Shared tier is supported.
+You can scale from S2 to S12 and vice versa. It is also possible to upgrade
+from Shared plans to Dedicated plans, but not the other way around. Snapshots
```
Comment (SStorm) [inline docs/reference/subscription-plans.rst:103]: Do we use the term "snapshots" or backups in other places? Just for consistency.

**Review by SStorm — APPROVED** [review]

File: docs/reference/subscription-plans.rst (hunk 3 above)
Comment (proddata) [reply docs/reference/subscription-plans.rst:94]: storage resources are not shared, they are dedicated

Comment (proddata) [reply docs/reference/subscription-plans.rst:94]:
```suggestion
Shared-resource plans are used to deploy clusters that share CPU resources with other clusters in this category. Memory and storage are allocated exclusively to each individual cluster.
```

**Comment by matkuliak** [comment]
> Hi @matkuliak
> 
> With the restructuring of the docs I think it is worth to have a top-level service offerings page, which I started here: #35 Let's bring this two pages together.

Hi @proddata. Sure, that makes sense. Should https://github.com/crate/cloud-docs/pull/36 replace the current [subscription plans](https://cratedb.com/docs/cloud/en/latest/reference/subscription-plans.html) page?
I can verify that all the info is there and then remove the https://cratedb.com/docs/cloud/en/latest/reference/subscription-plans.html

**Comment by matkuliak** [comment]
> Hi @matkuliak
> 
> With the restructuring of the docs I think it is worth to have a top-level service offerings page, which I started here: #35 Let's bring this two pages together.

Hi @proddata. Sure, that makes sense. Should https://github.com/crate/cloud-docs/pull/36 replace the current [subscription plans](https://cratedb.com/docs/cloud/en/latest/reference/subscription-plans.html) page?
I can verify that all the info is there and then remove the https://cratedb.com/docs/cloud/en/latest/reference/subscription-plans.html

**Pull request merged**